src/docks/encodedock.cpp

```diff
@@ -2042,3 +2047,38 @@ bool EncodeDock::detectHardwareEncoders()
     }
     return !hwlist.isEmpty();
 }
+
+bool EncodeDock::checkForMissingFiles()
+{
+    Mlt::Producer* service = fromProducer();
+    if (!service) {
+        service = MAIN.playlist();
+    }
+    if (!service) {
+        LOG_ERROR() << "Encode: No service to encode";
+        return true;
+    }
+    QTemporaryFile tmp;
```

Comment:
Use `QScopedPointer<QTemporaryFile> tmp(Util::writableTemporaryFile(filename));` where possible. 
Since you do not yet have a filename you could move this to after the file save dialog, or maybe use MAIN.filename() if not empty.

Reply:
Can you explain why a QScopedPointer is helpful here? The QTemporaryFile will be destroyed when the function exits.

Can you also explain the purpose of Util::writableTemporaryFile(filename)?

> Since you do not yet have a filename you could move this to after the file save dialog

My goal was to not even bother to show the file save dialog if there are missing files. It would be an unnecessary step.

Reply:
I saw the `tmp.remove()` and thought it might invalidate the auto-removal, but I think you are correct.

Reply:
The purpose of `Util::writableTemporaryFile(filename)` is to fix a bug where some user's system temp dir was not writable. Maybe someone does that to prevent saving personal info on a shared or public computer. I do not know. Or perhaps it is faulty setup. Nevertheless, it was found troubleshooting a bug report. In that case, the method creates the temp file in the same folder as `filename` where and usually filename was already checked with `Util::warnIfNotWritable()`.

The scoped pointer is used because the method returns a pointer to a new QTemporaryFile and does not destroy it.

Reply:
Thanks for the explanation. This is done.

Reply:
I did some testing, and when MAIN.fileName() is empty because it is an unsaved project, then it uses "." for the directory of the temp file when $TMPDIR (%TEMP% on Windows) is not writable. This ends up trying to use the location of the shotcut binary executable, which is usually an unwritable location.  Also, the temp file was not auto-removed. I will fix it in a commit after merge with
```
--- src/docks/encodedock.cpp
+++ src/docks/encodedock.cpp
@@ -2059,5 +2059,10 @@ bool EncodeDock::checkForMissingFiles()
         return true;
     }
-    QScopedPointer<QTemporaryFile> tmp{Util::writableTemporaryFile(MAIN.fileName())};
+    QScopedPointer<QTemporaryFile> tmp;
+    if (MAIN.fileName().isEmpty()) {
+        tmp.reset(Util::writableTemporaryFile(QStandardPaths::writableLocation(QStandardPaths::AppDataLocation) + "/"));
+    } else {
+        tmp.reset(Util::writableTemporaryFile(MAIN.fileName()));
+    }
     tmp->open();
     tmp->close();
@@ -2080,6 +2085,8 @@ bool EncodeDock::checkForMissingFiles()
         dialog.setDefaultButton(QMessageBox::Ok);
         dialog.exec();
+        QFile::remove(fileName);
         return true;
     }
+    QFile::remove(fileName);
     return false;
 }
```